dask/typing.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import Any, Callable, Mapping, MutableMapping, Protocol, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = object
+
+
+@runtime_checkable
+class DaskCollection(Protocol):
+    def __dask_graph__(self) -> Mapping | None:
```

Comment:
I was taking inspiration from the [implementation of `is_dask_collection`](https://docs.dask.org/en/stable/_modules/dask/base.html#is_dask_collection) checking for `is not None` instead of something like `hasattr`, but since you've pointed it out I feel that anyone implementing a collection would not have a `None` returning `__dask_graph__`.. So yea I'd agree it would make sense to just have `Mapping` here

Reply:
I'd be in favor of removing the possibility of `None`. I can't think of a scenario where it would pop up.

Reply:
I think if we are removing this, it's worth also deprecating/removing it in `is_dask_collection()`.